### PR TITLE
Debian10 automated DEBs build

### PR DESCRIPTION
**Pull request by hbokh**
This will add automated Debian 10 "Buster" DEB package builds to the CI pipeline.
Far from perfect due to issues in multiple places, eg. in `setup.py`; the `cobbler.spec`; etc.
Look at ` tests/build-and-install-debs.sh` and in the Travis buildlogs for where to improve.

### CODE REVIEW

**Comment by hbokh** [comment]
The RPM builds are green in Travis (but failed) due to the fact that this PR does not have the fixed from my previous PR included. 
The ehr... "interesting" stuff is here: https://travis-ci.org/cobbler/cobbler/jobs/643116167#L1654-L1696

File: tests/build-and-install-debs.sh [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+# Build DEBs
+mkdir -p deb-build tmp
+mkdir -p deb-build/{BUILD,BUILDROOT,DEBS,SDEBS,SOURCES}
+docker run -ti -v "$PWD/deb-build:/usr/src/cobbler/deb-build" -v "$PWD/tmp:/var/tmp" "$IMAGE"
+
+# Launch container and install cobbler
+docker run -d --privileged -v /sys/fs/cgroup:/sys/fs/cgroup:ro --name cobbler -v "$PWD/deb-build:/usr/src/cobbler/deb-build" "$IMAGE" /lib/systemd/systemd --system
+
+# THIS NEEDS TO BE FIXED !! In setup.py or somewhere else.
```
Comment (hbokh) [inline tests/build-and-install-debs.sh]: From here I had to fix things by hand for now.
For example, I don't know how to overrule the usage of `self.root` pointing at `/usr/local` in `setup.py`, L320.
How to auto enable the modules & config for Apache2; create directories in `/etc/cobbler/`.

**Comment by hbokh** [comment]
FWIW I collected the `deb-tmp` files and part of the output of `debbuild` of a new build in a gist:
https://gist.github.com/hbokh/b396484885e84fe4f101284a56732408

File: tests/build-and-install-debs.sh (hunk 1 above)
Comment (SchoolGuy) [reply tests/build-and-install-debs.sh]: Okay so enabling modules of Apache2 is not the concern of the `setup.py` in my eyes. This is something an admin has to do because we don't want to mess with the webserver configuration automatically (imagine the installation fails due to some env problem and the webserver is then in an broken state). The etc directory and files are already created (at least partly) in the data files section in the `setup.py L584-613`.

Regarding `self.root`: Just use the option `--root`. The hint for this can be found in `setup.py L318`.

Comment (hbokh) [reply tests/build-and-install-debs.sh]: This is hard, because of this line:
https://github.com/cobbler/cobbler/blob/master/Makefile#L124
and `DESTDIR=/`

**Comment by hbokh** [comment]
The real issue IMHO is, is that the impression is made there are **usable** DEB packages available for installation. But these DEBs and Cobbler 3 are not (yet). I want to try to get that fixed or I have  to drop Cobbler on Debian and look for an alternative eg. FAI.
So yeah, I'm busy trying. 😃

File: tests/build-and-install-debs.sh (hunk 1 above)
Comment (SchoolGuy) [reply tests/build-and-install-debs.sh]: You can use the following to override this: https://www.gnu.org/software/make/manual/make.html#Overriding

**Comment by SchoolGuy** [comment]
@hbokh What needs to be done to have that impression. Because at least SUSE Manager is using Cobbler 3 in production environments to manage datacenters. Yes they are only using it as a tool but the core functionality should work.

**Comment by hbokh** [comment]
The Travis job for "Debs" somehow gets cancelled today.

**Comment by SchoolGuy** [comment]
@hbokh Yes because the rpm step is failing.

**Comment by SchoolGuy** [comment]
For now I triggered it manually.

File: tests/build-and-install-debs.sh (hunk 1 above)
Comment (hbokh) [reply tests/build-and-install-debs.sh]: Tried really hard but I am unable to overrule `@@install_scripts@@` and `@@install_data@@`. It used to be like this:

https://github.com/cobbler/cobbler/commit/9eb1f39df9d9c430cc30fadd271a6df69f21ce43#diff-ff3ee63545d5735761cd6925c830c669

but are there ANY distros that install packages in `/usr/local` by default, except for eg. FreeBSD (ports)?

**Comment by hbokh** [comment]
These lines remain puzzling:
https://travis-ci.org/cobbler/cobbler/jobs/644374259#L1682-L1685

File: tests/build-and-install-debs.sh (hunk 1 above)
Comment (SchoolGuy) [reply tests/build-and-install-debs.sh]: I would actually like to see this possible because then we can use the make file to perform user installations.

Comment (SchoolGuy) [reply tests/build-and-install-debs.sh]: But if you can't get it to work then this is also fine for me.

Comment (hbokh) [reply tests/build-and-install-debs.sh]: > But if you can't get it to work then this is also fine for me.

As stated, I am unable to overrule `@@install_scripts@@` and `@@install_data@@`.
I do not like my solution at all, but I can't get through the options of the `Makefile` and the `setup.py`... 😢
Looking into the `setup.py`, there must be a way to set `--install-scripts` & `--install-data` for the install option, but I don't get it.

Comment (hbokh) [reply tests/build-and-install-debs.sh]: To share some of my confusion: in `setup.py` setting `self.root = "/usr/local"` to `/usr` does not make any difference. So I wonder, why is it (still) there?

**Comment by hbokh** [comment]
I wonder, would it be possible to override this line?
https://travis-ci.org/cobbler/cobbler/jobs/644803844#L2250
It’s created by this debbuild macro: https://github.com/debbuild/debbuild-macros/blob/master/macros.python3
And if so, should it be overridden in the `Makefile`, the spec file or `setup.py`?

File: tests/build-and-install-debs.sh (hunk 1 above)
Comment (SchoolGuy) [reply tests/build-and-install-debs.sh]: F*** I have no idea and this means probably that this is broken anyway. But I don't think we can remove it because there is probably something depending on it during the installation process...

File: cobbler.spec [hunk 2]
```diff
@@ -306,6 +299,7 @@ fi
 %post
 %{py3_bytecompile_post %{name}}
 %{systemd_post cobblerd.service}
+%{apache2_module_post proxy_http}
```
Comment (Conan-Kudo) [inline cobbler.spec]: Please add `BuildRequires: apache2-deb-macros` in a debbuild vendor conditional

Comment (hbokh) [reply cobbler.spec]: Fixed.

File: cobbler.spec [hunk 3]
```diff
@@ -328,20 +326,16 @@ fi
...
 
 %files
 %license COPYING
 %doc AUTHORS.in README.md
 %doc docs/developer-guide.rst docs/quickstart-guide.rst docs/installation-guide.rst
-%if %{_vendor} == "debbuild"
-%config(noreplace) %{_sysconfdir}/cobbler/*
-%else
 %config(noreplace) %{_sysconfdir}/cobbler
-%endif
```
Comment (Conan-Kudo) [inline cobbler.spec]: config files cannot be directories... If you're going to do this, then enumerate all the config files and mark them as `%config(noreplace)`

File: cobbler/settings.py [hunk 4]
```diff
@@ -100,7 +100,7 @@
     "mgmt_parameters": [{}, "dict"],
     "next_server": ["127.0.0.1", "str"],
     "nsupdate_enabled": [0, "bool"],
-    "power_management_default_type": ["ipmitool", "str"],
+    "power_management_default_type": ["ipmilan", "str"],
```
Comment (Conan-Kudo) [inline cobbler/settings.py]: what is `ipmilan`? This seems like an unrelated change...

File: dockerfiles/Debian10.dockerfile [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    systemd && \
+    apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+
+# Install debbuild packages (not in any repository)
+WORKDIR /var/cache/apt/archives
+RUN wget https://github.com/debbuild/debbuild/releases/download/$DBVERSION/debbuild_$DBVERSION-0ubuntu18.04_all.deb && \
+    wget https://github.com/debbuild/debbuild-macros/releases/download/0.0.2/debbuild-macros_0.0.2-0ubuntu18.04_all.deb  && \
+    dpkg -i debbuild_$DBVERSION-0ubuntu18.04_all.deb && \
+    dpkg -i debbuild-macros_0.0.2-0ubuntu18.04_all.deb && \
+    rm -f \*.deb
```
Comment (Conan-Kudo) [inline dockerfiles/Debian10.dockerfile]: Wouldn't this repository count? http://download.opensuse.org/repositories/Debian:/debbuild/Debian_10/

File: dockerfiles/Debian10.dockerfile [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+# Make /bin/sh point to bash, not dash
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+RUN echo "dash dash/sh boolean false" | debconf-set-selections && \
+    dpkg-reconfigure dash
```
Comment (Conan-Kudo) [inline dockerfiles/Debian10.dockerfile]: This should be unnecessary now.

File: docs/conf.py [hunk 7]
```diff
@@ -47,7 +47,6 @@
     'sphinx.ext.mathjax',
     'sphinx.ext.ifconfig',
     'sphinx.ext.viewcode',
-    'sphinx.ext.githubpages',
```
Comment (Conan-Kudo) [inline docs/conf.py]: This will break production of the cobbler docs publishing...

File: cobbler/settings.py (hunk 4 above)
Comment (hbokh) [reply cobbler/settings.py]: It is indeed unrelated to this PR, but it keeps on popping up while testing.
See https://gitter.im/cobbler/community?at=5e32e80715941335583ee67a

File: dockerfiles/Debian10.dockerfile (hunk 6 above)
Comment (hbokh) [reply dockerfiles/Debian10.dockerfile]: Removed.

File: docs/conf.py (hunk 7 above)
Comment (hbokh) [reply docs/conf.py]: No, but `sphinx.ext.githubpages` will break the build for CentOS7 due to an outdated version of Sphinx-docs. 
From @SchoolGuy:
>  I just looked at the github-pages-plugin of Sphinx and we don't need that because we don't publish the docs on GH-Pages. So we can remove this without any side effects in my eyes.

File: dockerfiles/Debian10.dockerfile (hunk 5 above)
Comment (hbokh) [reply dockerfiles/Debian10.dockerfile]: It does. Only learned from its existence yesterday.
I have update the DL location to the packages and mentioned it.
Setting up the repo with release keys etc. is too much of a hassle (for now) and will only increase the container image.

File: cobbler.spec (hunk 3 above)
Comment (hbokh) [reply cobbler.spec]: Reverted to Enno's change.
Will result in some more `dpkg-deb: warning: conffile`.

Comment (hbokh) [reply cobbler.spec]:
```console
dpkg-deb: warning: conffile '/etc/cobbler/boot_loader_conf' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/iso' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/power' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/reporting' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/settings.d' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/zone_templates' is not a plain file
dpkg-deb: warning: ignoring 6 warnings about the control file(s)
```

File: dockerfiles/Debian10.dockerfile (hunk 6 above)
Comment (hbokh) [reply dockerfiles/Debian10.dockerfile]: Not good judging these lines again:

```
sh: 1: Syntax error: "(" unexpected
sh: 1: Syntax error: "(" unexpected
```

File: cobbler/settings.py (hunk 4 above)
Comment (SchoolGuy) [reply cobbler/settings.py]: This is correct. This change is lgtm.

**Comment by SchoolGuy** [comment]
@hbokh Would you mind prefixen the PR with WIP and removing it when you think it is ready. My finger now went a few times over the merge button but I was not sure and then you pushed somehting again... (Also I would like to see if we could clean the git history when finished to remove the experiments.)

**Comment by hbokh** [comment]
@SchoolGuy Changed the title to include "WIP".
I am at the point where I personally cannot improve that much anymore, but lets keep it "In Progress" for a couple of days.

**Comment by hbokh** [comment]
Hi @SchoolGuy & @Conan-Kudo - this is stopping me from going on.

Using the DEBs build through Docker, installing in a fresh Vagrantbox:
```
root@cobbler:/vagrant# dpkg -i cobbler_3.1.2-1debian10_all.deb
Selecting previously unselected package cobbler.
(Reading database ... 31006 files and directories currently installed.)
Preparing to unpack cobbler_3.1.2-1debian10_all.deb ...
Unpacking cobbler (3.1.2-1debian10) ...
dpkg: error processing archive cobbler_3.1.2-1debian10_all.deb (--install):
 unable to create '/etc/cobbler/boot_loader_conf/bootcfg_esxi5.template.dpkg-new' (while processing './etc/cobbler/boot_loader_conf/bootcfg_esxi5.template'): No such file or directory
dpkg-deb: error: paste subprocess was killed by signal (Broken pipe)
Errors were encountered while processing:
 cobbler_3.1.2-1debian10_all.deb
```

I have no clue how to get around this. In the spec-file? In `setup.py`?
And how about the other warnings during the build?

```
dpkg-deb: warning: conffile '/etc/cobbler/boot_loader_conf' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/iso' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/power' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/reporting' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/settings.d' is not a plain file
dpkg-deb: warning: conffile '/etc/cobbler/zone_templates' is not a plain file
dpkg-deb: warning: ignoring 6 warnings about the control file(s)
```

If this cannot be fixed, these packages are almost useless.

**Comment by SchoolGuy** [comment]
@hbokh Investigating as soon as I have time (or cant/wont sleep).

**Comment by hbokh** [comment]
FYI In the meantime I have reverted to the regular way of building Debian packages, so I can build installable packages for the time being.
The Debian packages for cobbler & cobbler-web have these conffiles (checked with `dpkg -s cobbler` when build from release 3.1.1):

```txt
cobbler - Conffiles:

/etc/apache2/conf-available/cobbler.conf
/etc/cobbler/auth.conf
/etc/cobbler/boot_loader_conf/bootcfg_esxi5.template
/etc/cobbler/boot_loader_conf/bootcfg_esxi51.template
/etc/cobbler/boot_loader_conf/bootcfg_esxi55.template
/etc/cobbler/boot_loader_conf/bootcfg_esxi60.template
/etc/cobbler/boot_loader_conf/bootcfg_esxi65.template
/etc/cobbler/boot_loader_conf/bootcfg_esxi67.template
/etc/cobbler/boot_loader_conf/gpxe_system_esxi5.template
/etc/cobbler/boot_loader_conf/gpxe_system_esxi6.template
/etc/cobbler/boot_loader_conf/gpxe_system_freebsd.template
/etc/cobbler/boot_loader_conf/gpxe_system_linux.template
/etc/cobbler/boot_loader_conf/gpxe_system_local.template
/etc/cobbler/boot_loader_conf/gpxe_system_windows.template
/etc/cobbler/boot_loader_conf/grublocal.template
/etc/cobbler/boot_loader_conf/grubprofile.template
/etc/cobbler/boot_loader_conf/grubsystem.template
/etc/cobbler/boot_loader_conf/pxedefault.template
/etc/cobbler/boot_loader_conf/pxelocal.template
/etc/cobbler/boot_loader_conf/pxelocal_ia64.template
/etc/cobbler/boot_loader_conf/pxeprofile.template
/etc/cobbler/boot_loader_conf/pxeprofile_arm.template
/etc/cobbler/boot_loader_conf/pxeprofile_esxi.template
/etc/cobbler/boot_loader_conf/pxesystem.template
/etc/cobbler/boot_loader_conf/pxesystem_arm.template
/etc/cobbler/boot_loader_conf/pxesystem_esxi.template
/etc/cobbler/boot_loader_conf/pxesystem_ia64.template
/etc/cobbler/boot_loader_conf/pxesystem_ppc.template
/etc/cobbler/boot_loader_conf/yaboot_ppc.template
/etc/cobbler/cheetah_macros
/etc/cobbler/cobbler.conf
/etc/cobbler/cobbler_web.conf
/etc/cobbler/cobblerd.service
/etc/cobbler/cobblerd_rotate
/etc/cobbler/dhcp.template
/etc/cobbler/dnsmasq.template
/etc/cobbler/genders.template
/etc/cobbler/import_rsync_whitelist
/etc/cobbler/iso/buildiso.template
/etc/cobbler/logging_config.conf
/etc/cobbler/modules.conf
/etc/cobbler/mongodb.conf
/etc/cobbler/named.template
/etc/cobbler/ndjbdns.template
/etc/cobbler/reporting/build_report_email.template
/etc/cobbler/rsync.exclude
/etc/cobbler/rsync.template
/etc/cobbler/secondary.template
/etc/cobbler/settings
/etc/cobbler/settings.d/bind_manage_ipmi.settings
/etc/cobbler/settings.d/manage_genders.settings
/etc/cobbler/settings.d/nsupdate.settings
/etc/cobbler/users.conf
/etc/cobbler/users.digest
/etc/cobbler/version
/etc/cobbler/zone.template

cobbler-web - Conffiles:

/etc/apache2/conf-available/cobbler_web.conf
```

**Comment by Conan-Kudo** [comment]
@hbokh Thanks for that, I'm working on making fixes for this. 👍

**Comment by SchoolGuy** [comment]
So where are we currently with this? @Conan-Kudo is preparing fixes for the conf files and probably in debbuild and your part is basically done after that @hbokh ?

**Comment by Conan-Kudo** [comment]
I'm hopefully going to have at least spec fixes proposed this week. debbuild fixes are slightly more complicated. :)

**Comment by hbokh** [comment]
> So where are we currently with this? @Conan-Kudo is preparing fixes for the conf files and probably in debbuild and your part is basically done after that @hbokh ?

Yes. **Basically**, indeed. Maybe not perfect, but better than what we have now.

**Comment by SchoolGuy** [comment]
I think we should then merge after @Conan-Kudo has done his fixes. I think we should work more on fixing cobbler itself for now then focusing 100% of our time into building it perfectly.

**Comment by SchoolGuy** [comment]
@Conan-Kudo How is your progess with the debbuild problems?

**Comment by Conan-Kudo** [comment]
@SchoolGuy I should have a PR up tomorrow, hopefully. 😄

**Comment by hbokh** [comment]
Rebased / squashed.

**Comment by hbokh** [comment]
Had to put some things back on track, like finally adding the debbuild repo into the Dockerfile (instead of downloading DEBs based on hard coded versions).

**Comment by SchoolGuy** [comment]
@hbokh Is this still a WIP or ready to merge? If ready then please adjust the title and I will merge this!

**Comment by hbokh** [comment]
> @hbokh Is this still a WIP or ready to merge? If ready then please adjust the title and I will merge this!

The configs  in `/etc/cobbler` for `debbuild` are the last to be fixed AFAICS. But that's being worked on by @Conan-Kudo.

**Comment by Conan-Kudo** [comment]
Let me get that prepared first before we get this in. :)

**Comment by Conan-Kudo** [comment]
@hbokh @SchoolGuy Submitted as #2292.

**Comment by SchoolGuy** [comment]
@hbokh Go for a rebase! PR is merged!

File: dockerfiles/CentOS7.dockerfile [hunk 8]
```diff
@@ -29,6 +29,10 @@ RUN yum install -y          \
     python36-sphinx         \
     rpm-build
 
+# TEMPORARY !! This package is in EPEL testing since 20200127
+# These lines can be deleted once the package has passed.
+RUN rpm -ihv https://kojipkgs.fedoraproject.org/packages/python3-cheetah/3.2.4/1.el7/x86_64/python36-cheetah-3.2.4-1.el7.x86_64.rpm
```
Comment (Conan-Kudo) [inline dockerfiles/CentOS7.dockerfile]: This can be dropped now, it's now in EPEL 7.

**Comment by hbokh** [comment]
A the fail whale... 🐳

```
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/dhcp.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/dnsmasq.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/genders.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/iso/buildiso.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/named.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/ndjbdns.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/rsync.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/secondary.template': No such file or directory
/bin/cp: cannot stat '/usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/etc/cobbler/zone.template': No such file or directory
```

File: dockerfiles/CentOS7.dockerfile (hunk 8 above)
Comment (hbokh) [reply dockerfiles/CentOS7.dockerfile]: That is not supposed to be in here anymore.
Will have to look where I added old code...

**Comment by hbokh** [comment]
It's only the files from `cobbler/deb-build/cobbler-3.1.2/templates` with the template extension that fail.
In  `iso\` & `etc\` only, NOT the ones that come from `reporting/` and `boot_loader_conf/`.
Puzzling.

**Comment by hbokh** [comment]
Looks like the issue is in `setup.py`, not `cobbler.spec`.
Another force push and see where this goes.

This one is still here:

```
sh: -c: line 0: `/bin/cp -ar /usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64%attr(-,www-data,www-data) /usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/cobbler-web% '
sh: -c: line 0: syntax error near unexpected token `('
sh: -c: line 0: `/bin/cp -ar /usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64%attr(-,www-data,www-data) /usr/src/cobbler/deb-build/BUILDROOT/cobbler-3.1.2-1debian10.amd64/cobbler-web% '
```

**Comment by Conan-Kudo** [comment]
@hbokh Whoops, I forgot about that. I'll send a PR later today on that!

**Comment by hbokh** [comment]
> @hbokh Whoops, I forgot about that. I'll send a PR later today on that!

Forgot about `setup.py` or that www-data thing?
Travis is green now.

**Comment by Conan-Kudo** [comment]
@hbokh the www-data thing.

File: cobbler/settings.py (hunk 4 above)
Comment (hbokh) [reply cobbler/settings.py]: Reverted because of #2267

**Comment by hbokh** [comment]
I've been breaking my balls to get things the same in Debian 10 as in Fedora 31.
Next push will have:

- add missing directory `/etc/cobbler/power/` (incl. power templates that where removed for reasons unknown, yet there is `settings:power_template_dir: "/etc/cobbler/power"`)
- add missing directory `/etc/zone_templates`, with a `foo.example.com` as placeholder, since it seems to be impossible to create empty dirs with debbuild.

**Comment by hbokh** [comment]
> @hbokh Whoops, I forgot about that. I'll send a PR later today on that!

Any progress, @Conan-Kudo ?

**Comment by Conan-Kudo** [comment]
@hbokh Sorry, I got side-tracked by other things. I've got it on my list for this afternoon (I'm stuck in meetings all morning!).

**Comment by Conan-Kudo** [comment]
PR submitted as #2300.

**Comment by SchoolGuy** [comment]
@hbokh Merged so you can rebase. Hopefully then we can get this merged!

**Comment by hbokh** [comment]
> @hbokh Merged so you can rebase. Hopefully then we can get this merged!

Rebased & force pushed.

**Comment by SchoolGuy** [comment]
I looked through the log and would say we can merge. The only things which are probably a look worth are the following line: https://travis-ci.org/cobbler/cobbler/jobs/659567678#L1693-L1696

````
 * Warning:  git-core is probably installed but seems to be a virtual package.
 * Warning:  python3-deb-macros is probably installed but seems to be a virtual package.
 * Warning:  apache2-deb-macros is probably installed but seems to be a virtual package.
 * Warning:  systemd-deb-macros is probably installed but seems to be a virtual package.
````

If you agree @hbokh I would say you have the honor of removing the WIP prefix of the PR!

**Comment by Conan-Kudo** [comment]
@SchoolGuy Those are fine and expected.

File: .travis.yml [hunk 9]
```diff
@@ -34,6 +34,9 @@ jobs:
     - stage: rpms
       name: "Build Fedora 31 RPMs"
       script: ./tests/build-and-install-rpms.sh fc31 dockerfiles/Fedora31.dockerfile
+    - stage: packages
```
Comment (Conan-Kudo) [inline .travis.yml]: Can we name this stage `debs` instead of `packages`?

Comment (hbokh) [reply .travis.yml]: Fine by me. Will change after finishing dinner.
